fendermint/vm/interpreter/src/fvm/genesis.rs

```diff
@@ -247,6 +235,24 @@ where
             )
             .context("failed to create chainmetadata actor")?;
 
+        let eam_state = fendermint_actor_eam::State::new(
+            state.store(),
+            // or we can allow certain address only with:
+            // fendermint_actor_eam::PermissionModeParams::AllowList(
+            //  vec![...]
+            // )
+            fendermint_actor_eam::PermissionModeParams::Unrestricted,
```

Comment:
Is there a ticket number to track that this has to go in the genesis file?

Reply:
yep, #706

Reply:
@aakoshh My view here is that we should be extremely selective with what goes in genesis, as it could easily lead to configuration bloat as we keep adding more custom actors, or more knobs. Since our early users will be forking Fendermint, I'd rather have them use whichever mechanism they want to set the policy here.

Reply:
My opinion is what can change should go into the config, rather than be hardcoded and force people to fork just to change a value. We obviously cannot leave it like this, it has to go into some config. Alternatively we should not do the replacement here at all, just farm out the ability to do so into the application root, to minimise the surface area of where users have to look if they fork or import the code. 

In general, things that affect the consensus should be in the `genesis.json` file, and other settings that can vary between nodes should live outside. This once clearly affects the state, so it's a thing for `genesis.json`. 

In the Comsos SDK for example, every module can have their own section in the genesis, it's just a JSON file after all, and which keys goes in there depends on the node configuration. Here's the [genesis for staking](https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/staking/v1beta1/genesis.proto) for example. 

I agree that we don't want bloat. For example I didn't put in the ability to disable the chain metadata, even though it affects consensus, because I think it won't be used outside tests. However, this particular actor is useless without configuration, and everyone has to agree on who is allowed to deploy. In this case sharing the genesis is easiest if it's nicely wrapped up in a single file, rather than that file, plus some other config that people have to remember to also send to their peers.

That said, I see that the fact we grab configuration for the genesis from on-chain complicates things to some extent, ie. not sure if we want this to be a configuration of the subnet creation CLI and the contracts. 

Anyway, I just wanted to point out that it has to go _somewhere_.

Reply:
@aakoshh the original intention was to offer this actor, but not to wire it in and ship it as a default actor in genesis (which is what this PR is doing). Keeping the code inchantations commented with some instructions is what I originally proposed here.

The rationale being that bloating genesis does not scale as we introduce additional IPC actors down the line: new FVM runtimes, new system services (tunnels, bridges, etc.), etc. We'll eventually move to a content-addressed manifest model that declares the contents of the subnet both at genesis, and at upgrade checkpoints.

But if you feel strongly about shipping with this modified actor as a default and exposing the config through genesis, I can live with that. More so because @cryptoAtwill had already done that and he dropped the change on my request.

Reply:
@cryptoAtwill let's just call the shots here -- @cryptoAtwill can you re-add the genesis config for this actor and make sure that it can tackle all three cases (unrestricted, restricted with nobody allowed, restricted with specific addresses allowed).